### PR TITLE
Remove algorithm.NodeLister from scheduler interface

### PR DESCRIPTION
**Pull request by mrkm4ntr**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug

/kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
Remove redundant argument algorithm.NodeLister from scheduler interface.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Review by draveness — COMMENTED** [review]
/ok-to-test
/cc @alculquicondor

**Comment by alculquicondor** [comment]
Now that we are using the cache's data, this change makes a lot of sense.

This could be the path to solve #80372

@mrkm4ntr  consider opening an issue if you are doing more cleanups like this.

/lgtm
/assign @Huang-Wei

**Comment by mrkm4ntr** [comment]
@alculquicondor Thank you for your review.

> @mrkm4ntr consider opening an issue if you are doing more cleanups like this.

OK, I'll do that. And I will also check this issue.
https://github.com/kubernetes/kubernetes/issues/68951

**Review by ahg-g — COMMENTED** [review]
you can also remove "scheduledPodLister corelisters.PodLister" from Configurator, I am fine if you remove it in this PR.

**Comment by alculquicondor** [comment]
> you can also remove "scheduledPodLister corelisters.PodLister" from Configurator, I am fine if you remove it in this PR.

+1
That's definitely a left over from my last refactoring :(

**Review by Huang-Wei — COMMENTED** [review]
It's neat! Thanks, @mrkm4ntr.

Could you also clean up other occurrences?

```
pkg/scheduler/internal/cache/debugger/debugger.go
36:	nodeLister corelisters.NodeLister,
43:			NodeLister: nodeLister,

pkg/scheduler/internal/cache/debugger/comparer.go
34:	NodeLister corelisters.NodeLister
40:// Compare compares the nodes and pods of NodeLister with Cache.Snapshot.
45:	nodes, err := c.NodeLister.List(labels.Everything())
```

And a nit comment below.

**Comment by mrkm4ntr** [comment]
@Huang-Wei 
> Could you also clean up other occurrences?
>
> pkg/scheduler/internal/cache/debugger/debugger.go
36:	nodeLister corelisters.NodeLister,
43:			NodeLister: nodeLister,
>
> pkg/scheduler/internal/cache/debugger/comparer.go
34:	NodeLister corelisters.NodeLister
40:// Compare compares the nodes and pods of NodeLister with Cache.Snapshot.
45:	nodes, err := c.NodeLister.List(labels.Everything())

Is it really no longer necessary to compare the lister node to the cached node?

**Comment by Huang-Wei** [comment]
> Is it really no longer necessary to compare the lister node to the cached node?

@mrkm4ntr They're for debugging purpose. We can keep them.

File: cmd/kube-scheduler/app/server.go [hunk 1]
```diff
@@ -172,7 +172,7 @@ func Run(cc schedulerserverconfig.CompletedConfig, stopCh <-chan struct{}, regis
 	// Create the scheduler.
 	sched, err := scheduler.New(cc.Client,
 		cc.InformerFactory.Core().V1().Nodes(),
-		cc.PodInformer,
+		cc.InformerFactory.Core().V1().Pods(),
```
Comment (Huang-Wei) [inline cmd/kube-scheduler/app/server.go]: It doesn't look right. Original PodInformer is customized to filter out Succeeded/Failed pods:

https://github.com/kubernetes/kubernetes/blob/46e58df837ad83febd3fa1330c07bc0d353d406f/pkg/scheduler/factory/factory.go#L539-L548

File: cmd/kube-scheduler/app/server.go [hunk 2]
```diff
@@ -234,7 +234,6 @@ func Run(cc schedulerserverconfig.CompletedConfig, stopCh <-chan struct{}, regis
 	}
 
 	// Start all informers.
-	go cc.PodInformer.Informer().Run(stopCh)
```
Comment (Huang-Wei) [inline cmd/kube-scheduler/app/server.go]: If you remove this and rely on cc.InformerFactory.Start(), you will get all pods update.

File: cmd/kube-scheduler/app/server.go (hunk 1 above)
Comment (mrkm4ntr) [reply cmd/kube-scheduler/app/server.go]: Oh, you are right. I'll fix it.

**Comment by Huang-Wei** [comment]
/approve

Please squash the commits and we will /lgtm afterward.

**Pull request merged**